### PR TITLE
ceph: convert CephClient CRD to controller-runtime

### PR DESCRIPTION
**Pull request by leseb**
**Description of your changes:**

This was the last remaining CRD to not use the controller-runtime
library.
Small additions were added with the transition:

* the Kubernetes Secret that contains the CephX key has now an owner
  reference to the CephClient object
* the secret name is present in the Status field of the CephClient:

```
status:
  info:
    secretName: rook-ceph-client-glance
  phase: Ready
```

The controller will reconcile on CR updates and also if the Kubernetes
Secret is deleted.

Closes: #4938

**Which issue is resolved by this Pull Request:**
Resolves https://github.com/rook/rook/issues/4938

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: pkg/operator/ceph/client/controller.go [hunk 1]
```diff
@@ -24,295 +24,346 @@ import (
...
-		caps = append(caps, name, cap)
+	// Populate clusterInfo during each reconcile
+	r.clusterInfo, _, _, err = mon.LoadClusterInfo(r.context, request.NamespacedName.Namespace)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "failed to populate cluster info")
 	}
 
-	return clientEntity, caps, nil
-}
+	// DELETE: the CR was deleted
```
Comment (travisn) [inline pkg/operator/ceph/client/controller.go:185]:
```suggestion
	// the CR was deleted
```

**Review by travisn — APPROVED** [review]
LGTM! just a tiny suggestion.

File: pkg/operator/ceph/client/controller.go (hunk 1 above)
Comment (leseb) [reply pkg/operator/ceph/client/controller.go:185]: It's the same pattern (line) in the other controllers :-)

Comment (travisn) [reply pkg/operator/ceph/client/controller.go:185]: that's fine, not sure why it stood out this time :)

**Pull request merged**